### ISSUE TITLE
[Staging 1341] Not getting any info at all about a room where we normally do unless i open and close UI and even then this is acting weird

### ISSUE THREAD

**Issue by Conchipedia** [StrangeLoopGames/EcoIssues#14428]
**Version:** 0.9.0.0 beta staging-1341
i dug into a cliff face leaving a 2 hole doorway space then hollowed it out a bit as i normally do
![snap5965](https://user-images.githubusercontent.com/29383891/72658024-34697380-39a3-11ea-80d5-5a1deffe1ef8.png)

No info about this being a room where it would normally show this
![snap5966](https://user-images.githubusercontent.com/29383891/72658036-582cb980-39a3-11ea-8af8-1bd5d7e14774.png)

now using this command
![snap5968](https://user-images.githubusercontent.com/29383891/72658052-93c78380-39a3-11ea-8ae8-c2deb44593f9.png)

i get the following Information in chat window
![snap5964](https://user-images.githubusercontent.com/29383891/72658056-a2159f80-39a3-11ea-86b2-63bb742188d8.png)
which is correct and is expected size of the natural room

now only part of this natural room is claimed by me shown by the green color of my claim
![snap5967](https://user-images.githubusercontent.com/29383891/72658070-d5f0c500-39a3-11ea-9793-8af93660813d.png)

also i get this weird information when i move into a non claimed part of the the room
![snap5969](https://user-images.githubusercontent.com/29383891/72658083-0c2e4480-39a4-11ea-9010-ccdbdb51009a.png)

now comes for the **interesting** bit...

if i stand in the unclaimed part of my room and Hide UI and then Reopen it it get this
![snap5970](https://user-images.githubusercontent.com/29383891/72658123-8232ab80-39a4-11ea-9de7-7ec808352867.png)

then if i move to the part of the room i own i get this
![snap5971](https://user-images.githubusercontent.com/29383891/72658142-c02fcf80-39a4-11ea-931b-ccc341534be4.png)

now if i step outside my Room i still get info about being inside a room
![snap5972](https://user-images.githubusercontent.com/29383891/72658174-21f03980-39a5-11ea-9b2f-53c09def5581.png)

and if i walk off my claim it shows this (which is expected)
![snap5973](https://user-images.githubusercontent.com/29383891/72658187-57952280-39a5-11ea-9274-244fd8543838.png)

now if i walk back onto my claim it now is shows i am back in that room
![snap5974](https://user-images.githubusercontent.com/29383891/72658223-d4280100-39a5-11ea-81ac-b8258fd4d973.png)

Closing and opening UI now shows Deed name
![snap5975](https://user-images.githubusercontent.com/29383891/72658255-439df080-39a6-11ea-920c-103c4cbcb871.png)

now if i enter back into the room and then close and open UI i get this
![snap5976](https://user-images.githubusercontent.com/29383891/72658267-6d571780-39a6-11ea-906c-26a77074ee03.png)

and then walk to part of the room i don't own i now get this
![snap5977](https://user-images.githubusercontent.com/29383891/72658274-85c73200-39a6-11ea-85d1-e628f6f0852c.png)

its been fun and games, but something is definitely not quite right with Visual updating of Room description.

**Comment by Conchipedia** [comment]
the only reason i noticed the stuff happening when opening and closing UI was by accident as wanted to show the Natural room without UI

**Comment by Conchipedia** [comment]
in [Staging 1355] closing and opening UI (ALT+Z) again doesn't seem to make anything reappear now, only way i can get room info back is to disconnect

**Comment by johnkslg** [comment]
Was only able to repro the bad sizing when toggling UI on and off, I fixed that.
The other issues with inaccurate data displayed I couldnt repro. I suspect there was some unrelated exception beforehand that broke it, if you get it again can you attach your log?
@beastyn @SlayksWood may want to verify no repro for the remaining issues after my fix goes in.

**Comment by Conchipedia** [comment]
loaded up my Old word this still giving me the weird info bottom right, here is my Log and world
[Player.log](https://github.com/StrangeLoopGames/EcoIssues/files/4124314/Player.log)
[Storage.zip](https://github.com/StrangeLoopGames/EcoIssues/files/4124315/Storage.zip)

**Comment by beastyn** [comment]
1377
Pretty easy to reproduce
/bt 4,10,4,10
/claim
and nothing
No deed, no room,
![bandicam 2020-01-30 16-59-49-541](https://user-images.githubusercontent.com/27898520/73448233-2d1e6e80-4382-11ea-99a8-8425502fc896.jpg)
![bandicam 2020-01-30 17-00-24-443](https://user-images.githubusercontent.com/27898520/73448240-2ee83200-4382-11ea-84c7-3473aaaf5cc1.jpg)
![bandicam 2020-01-30 17-01-21-197](https://user-images.githubusercontent.com/27898520/73448242-30195f00-4382-11ea-8b50-9d1d50f43a4d.jpg)

**Comment by johnkslg** [comment]
Still cant repro
On freshly spawned world, /bt 4,10,4,10, then flew into room:
![image](https://user-images.githubusercontent.com/3536496/74065596-1e4f4180-49aa-11ea-8044-8eae5c69b419.png)

/claim, then moved slightly:
![image](https://user-images.githubusercontent.com/3536496/74065613-28714000-49aa-11ea-85cf-962331f4554b.png)

Can you try again in latest and PM me if its still failing and we will work out a repro case.

**Comment by SlayksWood** [comment]
This still happens, but I can't find any reprosteps.

**Comment by Conchipedia** [comment]
[0.9.0 Staging 1392] Created A random Singleplayer World Put down Campsite nothing coming up that this is my Deed (the box changes size but no text)
![snap6223](https://user-images.githubusercontent.com/29383891/74253269-c6d10e80-4ce6-11ea-8217-792b9a94fcf5.png)
![snap6224](https://user-images.githubusercontent.com/29383891/74253284-c9cbff00-4ce6-11ea-9cc3-2aab8cb5a6be.png)
also when Claiming new land expecting to see it change in colour like it used to do, just getting a very Faint Change in like a shading box
![snap6225](https://user-images.githubusercontent.com/29383891/74253598-30e9b380-4ce7-11ea-9deb-fbf3cb946502.png)
![snap6226](https://user-images.githubusercontent.com/29383891/74253602-32b37700-4ce7-11ea-9997-6b24f7a21b1c.png)
and if you try to claim in bright light u can see nothing at all

**Comment by Conchipedia** [comment]
this just changed so now i can see my Deed colour when claiming, did try changing colour of deed but nothing happened, all i did was go under water and come out and decided to claim some more land and now its showing up
![snap6227](https://user-images.githubusercontent.com/29383891/74254393-4dd2b680-4ce8-11ea-875f-86abe372252a.png)
![snap6228](https://user-images.githubusercontent.com/29383891/74254399-50351080-4ce8-11ea-8125-11a65e799364.png)

**Comment by Conchipedia** [comment]
now gone back to not showing deed colour
![snap6234](https://user-images.githubusercontent.com/29383891/74256250-fbdf6000-4cea-11ea-9638-9bd489a5e574.png)

**Comment by beastyn** [comment]
@Conchipedia we have separate issue about colors #14667
For name to appear you need to make a move. Quite annoying though.

**Comment by SlayksWood** [comment]
There are no 100% steps to reproduce, this happens all the time during the game with many players.

**Comment by SlayksWood** [comment]
It happens with Long Deed Names. I have a short nick, so it's harder to understand. Step to reproduce:
- build room (/bt 2,20,7,20,1,1).
- enter room and claim 1 deed inside:
![image](https://user-images.githubusercontent.com/45708377/76495767-9b0d8980-6448-11ea-85b6-7334a1f2bb40.png)
- open deed setting and rename it(add more characters):
![image](https://user-images.githubusercontent.com/45708377/76495820-b2e50d80-6448-11ea-9026-8853e0354fbe.png)
- It's broken now:
![image](https://user-images.githubusercontent.com/45708377/76495860-c7290a80-6448-11ea-9f29-29829cccbcaa.png)

**Comment by Conchipedia** [comment]
so that's why. my deed is always called **Diggy Diggy Hole**
unless i'm on the Test Server and it's **Diggy Diggy Test Hole** 😄

**Comment by SlayksWood** [comment]
from #15135
> ![image](https://user-images.githubusercontent.com/61889138/76689154-fa2af400-662a-11ea-8dce-f97272dedfdc.png)
> 
> ![image](https://user-images.githubusercontent.com/61889138/76689170-1d55a380-662b-11ea-9a67-c593fdd256de.png)
> ![image](https://user-images.githubusercontent.com/61889138/76689192-4e35d880-662b-11ea-8eaa-6ae9916fe954.png)
> 
> step to reproduce.
> Be in a district and start pulling a cart, the "Dirt road x1" or whose property it is kind of message is then blocked by the district map info

**Comment by johnkslg** [comment]
Should be fixed, try it after this PR

**Comment by Conchipedia** [comment]
hoping it comes to testing server soon 🙏

**Comment by johnkslg** [comment]
This broke in a merge, fixing again.

**Comment by SlayksWood** [comment]
Check 2 ways to reproduce, seems fixed. I'll continue to check it.
